gui/src-tauri/src/main.rs

```diff
@@ -20,8 +21,19 @@ use crate::util::{
 	get_launch_path, show_error, valid_java_paths, Cli, JAVA_BIN, MINIMUM_JAVA_VERSION,
 };
 
+mod state;
 mod util;
 
+#[tauri::command]
+fn update_window_state(
+	window: tauri::Window,
+	state: tauri::State<Mutex<state::WindowState>>,
+) -> Result<(), String> {
+	let mut lock = state.lock().unwrap();
+	lock.update_state(&window).map_err(|err| err.to_string())?;
```

Comment:
Once the other errors use `color_eyre::Result`, you can map the `eyre::Report` using `format!("{:?}", err)`

https://docs.rs/eyre/0.6.8/eyre/struct.Report.html#display-representations